frontend/src/__mocks__/mockPipelinePodK8sResource.ts

```diff
@@ -0,0 +1,618 @@
+import { KnownLabels, PodKind } from '~/k8sTypes';
+import { genUID } from '~/__mocks__/mockUtils';
+import { TolerationEffect, TolerationOperator } from '~/types';
+
+type MockResourceConfigType = {
+  user?: string;
+  name?: string;
+  namespace?: string;
+  isPending?: boolean;
+};
+
+export const mockPipelinePodK8sResource = ({
+  user = 'test-user',
+  name = 'test-pod',
+  namespace = 'test-project',
+}: MockResourceConfigType): PodKind => ({
+  kind: 'Pod',
+  apiVersion: 'project.openshift.io/v1',
+  metadata: {
+    name,
+    generateName: name,
+    namespace,
+    uid: genUID('pod'),
+    resourceVersion: '4800675',
+    creationTimestamp: '2023-02-14T22:06:45Z',
+    labels: {
+      app: name,
+      'controller-revision-hash': `${name}-5b68f78f58`,
+      'notebook-name': name,
+      [KnownLabels.DASHBOARD_RESOURCE]: 'true',
+      'opendatahub.io/odh-managed': 'true',
+      'opendatahub.io/user': user,
+      statefulset: name,
+      'statefulset.kubernetes.io/pod-name': name,
+    },
+    annotations: {
+      'openshift.io/scc': 'restricted-v2',
+      'seccomp.security.alpha.kubernetes.io/pod': 'runtime/default',
+    },
+    ownerReferences: [
+      {
+        apiVersion: 'apps/v1',
+        kind: 'StatefulSet',
+        name,
+        uid: genUID('statefulset'),
+        controller: true,
+        blockOwnerDeletion: true,
+      },
+    ],
+    managedFields: [],
+  },
+  spec: {
+    volumes: [
+      {
+        name,
+        persistentVolumeClaim: {
+          claimName: name,
+        },
+      },
+      {
+        name: 'oauth-config',
+        secret: {
+          secretName: `${name}-oauth-config`,
+        },
+      },
+      {
+        name: 'tls-certificates',
+        secret: {
+          secretName: `${name}-tls`,
+        },
+      },
+      {
+        name: 'kube-api-access-zzc98',
+      },
+    ],
+    containers: [
+      {
+        name: 'step-main',
+        image: 'python:alpine3.6',
+        env: [
+          {
+            name: 'SSL_CERT_DIR',
+            value:
+              '/tekton-custom-certs:/etc/ssl/certs:/etc/pki/tls/certs:/system/etc/security/cacerts',
+          },
+        ],
+        resources: {},
+        volumeMounts: [
+          {
+            name: 'tekton-internal-downward',
+            readOnly: true,
```

Comment:
`readOnly` does not exists in [VolumeMount type](https://github.com/opendatahub-io/odh-dashboard/blob/3d325b10357429f8fb08c5536386e79b65ebd350/frontend/src/types.ts#L588). Either remove all the reference of readOnly in this file or add this as a optional property in the type.